src/scripts.js

```diff
@@ -1,50 +1,150 @@
 import DestinationRepo from './DestinationRepo.js'
-import Trips from './Trips.js';
+import TripRepo from './TripRepo.js';
+import Trip from './Trip.js';
 import { getPromise, allData } from './apiCalls';
 
+//Query selectors ================================
+let upcomingTrips = document.querySelector('.upcoming-trips-container');
+let pastTrips = document.querySelector('.past-trips-container');
+let pendingTrips = document.querySelector('.pending-trips-container');
+let totalThisYear = document.querySelector('.total-amount');
+let topNav = document.querySelector('.top-nav');
+
 //Global Variables ===============================
 let currentUser = null;
 let usersTrips = null;
+let date = '2022/05/11';
+
 //Event Listeners ================================
 window.addEventListener('load', () => {
   allData.then(data => {
     let trips = data[0].trips;
     let destinations = data[1].destinations;
-    let travelers = data[2].travlers;
+    let travelers = data[2].travelers;
+    setInitialData(trips, travelers);
+    setUpInitialDisplay(destinations)
   }).catch(error => console.log(error));
 });
 
 //Data Functions =====================================
-const setUpTravelerRepo = (travelersArray) => {
-  let travelers = travelersArray.map((traveler) => {
-    return new Traveler(traveler);
-  })
-  let travelerRepo = new TravelerRepo(travelers);
-  getRandomUser(travelerRepo)
-  return travelerRepo
+const setInitialData = (trips, travelers) => {
+  setUpTravelerRepo(travelers);
+  findUsersTrips(currentUser.id, setUpTripsRepo(trips));
+  console.log(currentUser);
+  console.log(usersTrips);
+}
+
+const setUpInitialDisplay = (destinations) => {
+  console.log(destinations);
+  let repo = setUpDestinationsRepo(destinations);
+  setTripDestinations(repo);
+  getUpcomingTrips();
+  getPastTrips();
+  getPendingTrips();
+  calculateTravelCostThisYear();
+  welcomeUser();
+}
+
+const setTripDestinations = (repo) => {
+  usersTrips.forEach((trip) => {
+  let tripDestination = repo.findDestination(trip.destination);
+  trip.destination = tripDestination;
+});
 }
 
 const getRandomUser = (travelers) => {
   let randomIndex = Math.floor(Math.random() * travelers.travelers.length)
   currentUser = travelers.travelers[randomIndex];
 }
 
+const setUpTravelerRepo = (travelersArray) => {
+  let travelers = travelersArray.map((traveler) => {
+    return new Traveler(traveler);
+  })
+  let travelerRepo = new TravelerRepo(travelers);
+  getRandomUser(travelerRepo);
+  return travelerRepo;
+}
+
 const setUpTripsRepo = (trips) => {
-  return new Trips(trips);
+  return new TripRepo(trips);
 }
 
 const findUsersTrips = (userID, trips) => {
   usersTrips = trips.findAllTripsByUser(userID);
+  let newTrips = usersTrips.map((trip) => {
+    return new Trip(trip);
+  });
+  usersTrips = newTrips;
 }
 
-const setUpDestinationsRepo = (destinations) {
+const setUpDestinationsRepo = (destinations) => {
   let destinationsArray = destinations.map((destination) => {
```

Comment:
Not sure what best practice is! But I did this map inside of my repository constructors. That way when you call .destinations it already returns destination instances. Then you would only have to import and use the Destination Repo in your scripts.js.